datasets/nasa-nex-gddp-cmip6/nasa_nex_gddp_cmip6.py

```diff
@@ -0,0 +1,178 @@
+from typing import List, Union
+import planetary_computer
+from collections import namedtuple
+
+import io
+import pystac
+
+from pctasks.core.models.task import WaitTaskResult
+from pctasks.core.storage import StorageFactory
+from pctasks.dataset.collection import Collection
+
+from operator import itemgetter
+from string import ascii_lowercase
+import itertools
+
+from kerchunk.combine import MultiZarrToZarr
+from kerchunk.hdf import SingleHdf5ToZarr
+from kerchunk.netCDF3 import NetCDF3ToZarr
+from kerchunk.utils import rename_target
+
+import adlfs
+import xarray as xr
+import xstac
+
+class Cmip6Collection(Collection):
+    def get_item(cls, paths, range_compress=False):
+        credential = planetary_computer.sas.get_token("nasagddp", "nex-gddp-cmip6").token
+        fs = adlfs.AzureBlobFileSystem("nasagddp", credential=credential)
+        template = pystac.Item(
+            "item",
+            geometry={
+                "type": "Polygon",
+                "coordinates": [
+                    [
+                        [180.0, -90.0],
+                        [180.0, 90.0],
+                        [-180.0, 90.0],
+                        [-180.0, -90.0],
+                        [180.0, -90.0],
+                    ]
+                ],
+            },
+            bbox=[-180, -90, 180, 90],
+            datetime=None,
+            properties={"start_datetime": None, "end_datetime": None},
+        )
+
+        datasets = [xr.open_dataset(fs.open(path)) for path in paths]
+        ds = xr.merge(datasets)
+        r = xstac.xarray_to_stac(ds, template, reference_system=4326)
+        r.id = '.'.join(itemgetter(0,1,3)(cls.path_to_item_id(cls, paths[0]).split('.')))
```

Comment:
Are these the same IDs as we currently have?

Reply:
Yes. Here is an example: 
Current PC: https://planetarycomputer.microsoft.com/api/stac/v1/collections/nasa-nex-gddp-cmip6/items/NorESM2-LM.ssp585.2100
Updated: https://pct-apis-staging.westeurope.cloudapp.azure.com/stac/collections/nasa-nex-gddp-cmip6/items/NorESM2-LM.ssp585.2100

Reply:
Yes. Examples:
https://planetarycomputer.microsoft.com/api/stac/v1/collections/nasa-nex-gddp-cmip6/items/NorESM2-LM.ssp585.2100
https://pct-apis-staging.westeurope.cloudapp.azure.com/stac/collections/nasa-nex-gddp-cmip6/items/NorESM2-LM.ssp585.2100